ivy/data_classes/array/experimental/norms.py

```diff
@@ -39,93 +39,149 @@ def l2_normalize(
         """
         return ivy.l2_normalize(self, axis=axis, out=out)
 
+    def batch_norm(
+        self: Union[ivy.NativeArray, ivy.Array],
+        mean: Union[ivy.NativeArray, ivy.Array],
+        variance: Union[ivy.NativeArray, ivy.Array],
+        /,
+        *,
+        offset: Optional[Union[ivy.NativeArray, ivy.Array]] = None,
+        scale: Optional[Union[ivy.NativeArray, ivy.Array]] = None,
+        training: bool = False,
+        eps: float = 1e-5,
+        momentum: float = 1e-1,
+        out: Optional[Tuple[ivy.Array, ivy.Array, ivy.Array]] = None,
+    ) -> ivy.Array:
+        """
+            ivy.Array instance method variant of ivy.batch_norm. This method
+            simply wraps the function, and so the docstring for ivy.batch_norm
+            also applies to this method with minimal changes.
+        Parameters
+        ----------
+        x
+            Input array of shape (N, *S, C), where N is the batch dimension,
+            *S corresponds to any number of spatial dimensions and
+             C corresponds to the channel dimension.
+        mean
+            Mean array used for input's normalization. If ``training=True``
+            then it must be one dimensional with size equal to the size of
+            channel dimension C. If ``training=False`` then it can be of any
+            shape broadcastble to the input shape.
+        variance
+            Variance array for the input's normalization. If ``training=True``
+            then it must be one dimensional with size equal to the size of
+            channel dimension C. If ``training=False`` then it can be of any shape
+            broadcastble to the input shape.
+        offset
+            An offset array. If present, will be added to the normalized input.
+            If ``training=True`` then it must be one dimensional with size equal
+            to the size of channel dimension C. If ``training=False`` then it can
+            be of any shape broadcastble to the input shape.
+        scale
+            A scale array. If present, the scale is applied to the normalized input.
+            If ``training=True`` then it must be one dimensional with size equal to
+            the size of channel dimension C. If ``training=False`` then it can be of
+            any shape broadcastble to the input shape.
+        training
+            If true, calculate and use the mean and variance of `x`. Otherwise, use the
```

Comment:
If I understand correctly, for training=True the **_running_** mean and variance are used. Would be nice to update this in all the docstring to avoid confusion.

Reply:
When `training` is set to `True`,  `batch_mean` and `batch_variance` is used to normalize the input. In this case, running mean and running variance are updated using these batch statistics and the mean and variance passed as input as highlighted in the docstring here: https://github.com/unifyai/ivy/blob/ba4a10202bab8da6a4eccdbc6ff9384b1599b2f6/ivy/functional/ivy/experimental/norms.py#L72-L82
Should I copy this description to container and arrays methods too? I thought that we only write a brief description in array and container methods and redirect the user to the docstring of the functional API with this line "This method simply
        wraps the function, and so the docstring for ivy.batch_norm also applies
        to this method with minimal changes."

Reply:
Oh you're right. I didn't notice the explanation at the start of the docstring at all. I went straight to the argument definition. No, I don't think you need to update the array and container methods.